src/en/kunmangato/src/eu/kanade/tachiyomi/extension/en/kunmangato/KunMangaTo.kt

```diff
@@ -0,0 +1,213 @@
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val mangas = super.popularMangaParse(response).mangas.distinctBy { it.url }
+        return MangasPage(mangas, false)
+    }
+
+    override fun popularMangaSelector(): String = ".sidebar-box-popular article"
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        val a = element.selectFirst("a.manga")!!
+        return SManga.create().apply {
+            title = a.text()
+            url = a.attr("href").removePrefix(baseUrl)
+            thumbnail_url = element.selectFirst("img")!!.attr("src")
+        }
+    }
+
+    override fun popularMangaNextPageSelector(): String? = null
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val urlBuilder = "$baseUrl/search".toHttpUrl().newBuilder()
+            .addQueryParameter("term", query.trim())
+            .addQueryParameter("page", page.toString())
+
+        filters.filterIsInstance<UriPartFilter>().forEach { filter ->
+            urlBuilder.addQueryParameter(
+                filter.toQueryParam(),
+                filter.toUriPart(),
+            )
+        }
+
+        return GET(urlBuilder.build(), headers)
+    }
+
+    override fun searchMangaSelector(): String = "article .card-manga"
+
+    override fun searchMangaFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun searchMangaNextPageSelector(): String = "ul.pagination-primary a[rel=next]"
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val urlBuilder = "$baseUrl/latest".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+
+        return GET(urlBuilder.build(), headers)
+    }
+
+    override fun latestUpdatesSelector(): String = searchMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element) = searchMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector(): String = searchMangaNextPageSelector()
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
+            title = document.selectFirst(".page-heading")!!.text()
+            author = document.selectFirst("p.mb-1:nth-child(1)")!!.text().drop(9)
+            description = document.getElementById("manga-description")!!.text()
+            genre =
+                document.select("a.manga-genre").joinToString(", ") { element -> element.text() }
+            status =
+                if (document.select(".text-success").size > 0) SManga.COMPLETED else SManga.ONGOING
+            thumbnail_url = document.selectFirst("img.text-end")!!.attr("src")
+        }
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val chaptersDocument =
+            Jsoup.parse("<div>" + document.selectFirst(chapterListSelector())!!.attr("value") + "</div>")
```

Comment:
use `Jsoup.parseBodyFragment` instead.
(won't need to add div tag manually i think)